engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -448,6 +450,14 @@ public void startApplication() throws YarnException, IOException
       }
       String libJarsCsv = copyFromLocal(fs, appPath, localJarFiles.toArray(new String[]{}));
 
+      if (UserGroupInformation.isSecurityEnabled() && customKeystoreConfig != null && customKeystorePath != null) {
+        String[] customSSLConfArr = {customKeystoreConfig, customKeystorePath};
+        String customSSLFileConf = copyFromLocal(fs, appPath, customSSLConfArr);
+        String configFileName = (new java.io.File(customKeystoreConfig)).getName();
+        dag.getAttributes().put(Context.DAGContext.STRAM_HTTP_CUSTOM_CONFIG, configFileName);
```

Comment:
I disagree that _CUSTOM_ is superfluous or incorrect in this context.

Some background: a user wanted to enable SSL but the current Yarn+Apex implementation only allows (i.e. forces) the default Hadoop SSL keystore to be used (pls see the JIRA APEXCORE-711 and YARN-6457). This keystore file was only accessible from "root" but the Apex app was running as non-root. The second problem is that there is no guarantee that the default Hadoop keystore is available on all Haoop nodes where the Stram could potentially run. So we need a mechanism of not only using a non-default (i.e. custom) keystore but also a mechanism to distribute it to the destination app-master node (this PR).

Also there is a precedent for this. Look at the H2O page http://docs.h2o.ai/h2o/latest-stable/h2o-docs/security.html where they are trying to address the same or similar issue and it does mention the word custom in  "Here is an example of how to create your own self-signed Java Keystore (mykeystore.jks) with a custom keystore password (mypass) and how to run standalone H2O using your Keystore:"

Regarding STRAM_ too, I don't see anything wrong with it: STRAM is Streaming App Master and this attribute WILL be used only on Stram until/unless we decide to use the same config for other non-Stram uses.